aten/src/ATen/cuda/Array.h

```diff
@@ -7,7 +7,7 @@
 namespace at { namespace cuda {
 
 template <typename T, int size>
-struct Array {
+struct alignas(16) Array {
```

Comment:
@syed-ahmed sorry to request this, but can you guard this with a `#ifndef __HIP_PLATFORM_HCC__`? looks like rocm cannot handle alignas correctly. cc: @iotamudelta @bddppq

Reply:
@soumith I just checked w/ our compiler team. We should support `alignas()` correctly. Could the CI test failure be temporary?

Reply:
Seems like the alignas not working correctly for RoCM builds is real, even after @bddppq 's retest. I have added the ifdef for this struct for now.